### PR TITLE
23 Chat system homework #1 - Stanislav Ivanov

### PR DESCRIPTION
**Pull request by Stanislav04**

### CODE REVIEW

**Comment by arnaudoff** [comment]
Please drop the .idea files from the merge request - they are not needed

File: homeworks/23/chat-system/src/main/java/org/example/Client.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.Scanner;
+
+public class Client {
+    public static void main(String[] args) {
+        var scanner = new Scanner(System.in);
+        String command = "";
+
+        while (!command.equals("/quit")) {
```
Comment (AlexGechovski) [inline homeworks/23/chat-system/src/main/java/org/example/Client.java:13]: The "/quit" command doesn't end the process.

File: homeworks/23/chat-system/src/main/java/org/example/Client.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+            OutputStream out = socket.getOutputStream();
+            var output = new DataOutputStream(out);
+            InputStream in = socket.getInputStream();
+            var input = new BufferedReader(new InputStreamReader(in));
+
+            var userInput = new Scanner(System.in);
+            String command = "";
+
+            while (!command.equals("/quit")) {
```
Comment (AlexGechovski) [inline homeworks/23/chat-system/src/main/java/org/example/Client.java:44]: The way that you receive messages means that they are received only when pressing enter. You should use a new thread for reading.

File: homeworks/23/chat-system/src/main/java/org/example/Server.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        try (var serverSocket = new ServerSocket(port)){
+            System.out.println("Server is listening on port \033[33;3m" + port + "\033[0m");
+            // System.out.println("Server is listening on port " + port);
+
+            while (!serverSocket.isClosed()) {
+                Socket socket = serverSocket.accept();
+                System.out.println(socket.getInetAddress().getHostAddress() + ":" + socket.getPort() + " \033[32mconnected\033[0m");
+                // System.out.println(socket.getInetAddress().getHostAddress() + ":" + socket.getPort() + " connected");
+
+                new ServerThread(socket).start();
```
Comment (AlexGechovski) [inline homeworks/23/chat-system/src/main/java/org/example/Server.java:19]: Thread pool would've been a better choice

**Review by AlexGechovski — COMMENTED** [review]
The code is readable but isn't working correctly. The /quit command isn't working, the thread doesn't terminate. The bigger problem is that the clients have to escape user input whenever they want to receive messages from other users. I think that the grade should be 4 , because the chat app isn't working properly and isn't written well.

**Pull request merged**